### ISSUE TITLE
2.2.1 maybe incompatible with HPOS (1962)

### ISSUE THREAD

**Issue by xbb123** [woocommerce/woocommerce-paypal-payments#1628]
### Describe the Bug

PayPal gateway up to 2.2.0 worked just fine, with 2.21, once go to the order edit page, a 500 will be returned.

PHP Fatal error:  Uncaught TypeError: WooCommerce\PayPalCommerce\Subscription\SubscriptionModule::WooCommerce\PayPalCommerce\Subscription\{closure}(): Argument #2 ($post) must be of type WP_Post, Automattic\WooCommerce\Admin\Overrides\Order given, called in /usr/share/nginx/html/website/wp-includes/class-wp-hook.php on line 310 and defined in /usr/share/nginx/html/website/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-subscription/src/SubscriptionModule.php:230
Stack trace:
#0 /usr/share/nginx/html/website/wp-includes/class-wp-hook.php(310): WooCommerce\PayPalCommerce\Subscription\SubscriptionModule->WooCommerce\PayPalCommerce\Subscription\{closure}()
#1 /usr/share/nginx/html/website/wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters()
#2 /usr/share/nginx/html/website/wp-includes/plugin.php(517): WP_Hook->do_action()
#3 /usr/share/nginx/html/website/wp-content/plugins/woocommerce/src/Internal/Admin/Orders/Edit.php(159): do_action()
#4" while reading response header from upstream, client: IP, server: www.xxxxx.com, request: "GET /wp-admin/admin.php?page=wc-orders&action=edit&id=47789 HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "www.xxxxx.com", referrer: "https://www.xxxxx.com/wp-admin/admin.php?page=wc-orders"

### To Reproduce

1. Install PayPal 2.2.1
2. Go to Woocommerce Orders
3. Click on a order

### Environment

 - WordPress Version 6.3
 - WooCommerce Version 8.0.2
 - Plugin Version 2.2.1
 - Chrome 116.0.5845.111
 - AutomateWoo, AutomateWoo - Refer A Friend Add-on, WooCommerce Google Analytics Integration, WooCommerce Product Recommendations, WooCommerce Stripe Gateway, Yoast SEO
### Additional Details

<details><summary>System status</summary>

  ```
  <!--If applicable, paste the system status here. Please ensure you redact or remove any identifying information. -->
  ```

</details>

**Comment by CameronD73** [comment]
Agree.  the error "Argument 2 must be of type WP_Post" seems a bit of a give-away.
If I switch back to post-based orders then the crash does not occur, even if I leave HPOS enabled and synchronised.

The crash occurs on all orders, even if payment does not use Paypal in any way.

**Comment by InpsydeNiklas** [comment]
Hi @xbb123  @CameronD73, Thanks for the report and apologies, this was not supposed to happen. A fix package can be [downloaded from here](https://drive.google.com/uc?export=download&id=16GC-69a1JmA65HH0GIj4xOAQNL313hDP) until it was released as part of a 2.2.2 update, hopefully soon.
